src/engine/source/api/include/api/test/sessionManager.hpp

```diff
@@ -0,0 +1,130 @@
+#ifndef _API_TEST_SESSION_MANAGER_HPP
+#define _API_TEST_SESSION_MANAGER_HPP
+
+#include <atomic>
+#include <optional>
+#include <shared_mutex>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <error.hpp>
+
+#include "session.hpp"
+
+namespace api::sessionManager
+{
+
+constexpr auto FILTER_NAME_ERROR_MSG = "Filter '{}' is already assigned to a route '{}'";
+constexpr auto POLICY_NAME_ERROR_MSG = "Policy '{}' is already assigned to a route '{}'";
+constexpr auto ROUTE_NAME_ERROR_MSG = "Route name '{}' already exists";
+constexpr auto SESSION_ID_ERROR_MSG = "Session ID '{}' already exists";
+constexpr auto SESSION_NAME_ERROR_MSG = "Session name '{}' already exists";
+
+constexpr auto SESSION_DEBUG_MSG = "Session created: ID={}, Name={}, Creation Date={}, Policy Name={}, Filter Name={}, "
+                                   "Route Name={}, Life Span={}, Description='{}'\n";
+
+/**
+ * @brief Session Manager class.
+ */
+class SessionManager
+{
+private:
+    std::unordered_map<std::string, Session> m_activeSessions; ///< Map of active sessions
+    std::unordered_map<std::string, std::string> m_filterMap;  ///< Map of active filter
+    std::unordered_map<std::string, std::string> m_policyMap;  ///< Map of active policies
+
+    std::unordered_set<std::string> m_routeSet;                ///< Set of active routes
+    std::unordered_set<uint32_t> m_idSet;                      ///< Set of active filter
+
+    uint32_t m_sessionIDCounter;      ///< This counter is used to provide a unique ID to each session
+
+    std::shared_mutex m_sessionMutex; ///< Mutex to protect the sessions resources usage
+
+public:
+    /**
+     * @brief Construct a new Session Manager object
+     *
+     */
+    SessionManager()
+        : m_sessionIDCounter {0}
```

Comment:
It is not necessary to add the 'u' suffix to the value 0 when initializing `m_sessionIDCounter` as it is already of type `uint32_t`. The 'u' suffix is used to explicitly denote an unsigned integer literal.

In this case, since `m_sessionIDCounter is already declared as `uint32_t`, it is implicitly an unsigned integer, and initializing it with '0' is perfectly valid and appropriate, therefore the 'u' suffix is not required in this context.